components/top-nav/src/index.js

```diff
@@ -0,0 +1,238 @@
+    paddingBottom: 0,
+    ':hover': {
+      borderBottom: `1px solid ${WHITE}`,
+    },
+  },
+  [MEDIA_QUERIES.LARGESCREEN]: {
+    ' button': {
+      display: 'none',
+    },
+    width: '66.66%',
+    paddingLeft: SPACING.SCALE_3,
+    marginTop: 0,
+  },
+});
+
+const SearchWrapper = glamorous.div({
+  marginTop: SPACING.SCALE_1,
+  width: '50%',
+  [MEDIA_QUERIES.LARGESCREEN]: {
+    width: '100%',
+  },
+});
+
+const Ul = glamorous.ul(({
+  serviceTitle,
+}) => ({
+  display: 'flex',
+  flexWrap: 'wrap',
+  margin: 0,
+  padding: 0,
+  marginTop: SPACING.SCALE_1,
+  width: '100%',
+  [MEDIA_QUERIES.LARGESCREEN]: {
+    paddingTop: serviceTitle ? 0 : SPACING.SCALE_5,
+    width: 'auto',
+  },
+}));
+
+const Li = glamorous.li(({
+  active,
+}) => ({
+  flex: '1 0 50%',
+  width: '100%',
+  listStyleType: 'none',
+  margin: 0,
+  padding: '3px 0',
+  borderBottom: `1px solid ${PROPOSITION_BORDER}`,
+  [MEDIA_QUERIES.LARGESCREEN]: {
+    borderBottom: 0,
+    flex: 'none',
+    width: 'auto',
+    paddingRight: SPACING.SCALE_3,
+  },
+  ' a': {
+    color: active ? LIGHT_BLUE : WHITE,
+    textDecoration: 'none',
+    borderBottom: '1px solid transparent',
+    fontWeight: 700,
+    ':hover': {
+      borderBottomColor: active ? LIGHT_BLUE : WHITE,
+    },
+  },
+}));
+
+const TopNav = ({
+  active,
+  bgColor,
+  color,
+  company,
+  serviceTitle,
+  search,
+  children,
+  ...props
+}) => (
+  <React.Fragment>
+    <TopNavWrapper bgColor={bgColor} color={color} {...props}>
+      <TopNavInner>
+        <LogoSearchWrapper>
+          <Company>
+            {company}
+          </Company>
+          {search &&
+            <SearchWrapper>
+              {search}
+            </SearchWrapper>
+          }
+        </LogoSearchWrapper>
+        <RightHandSide>
+          {serviceTitle}
+          {children &&
+          <React.Fragment>
+            <Button mb={0}>menu</Button>
+            <Ul serviceTitle={serviceTitle}>
+              {children.length && children.map ? (
+                children.map((child, i) => (
+                  child && (child.length || child.props)
+                    ? <Li active={active === i ? 'active' : undefined} key={child.key || i}>{child}</Li>
+                    : null
+                ))
+              ) : (
+                <Li active={active}>{children}</Li>
+              )}
+            </Ul>
+          </React.Fragment>
+          }
+        </RightHandSide>
+      </TopNavInner>
+    </TopNavWrapper>
+    <BottomNavWrapper />
+  </React.Fragment>
+);
+
+TopNav.defaultProps = {
+  active: undefined,
+  bgColor: BLACK,
+  color: WHITE,
+  company: undefined,
+  serviceTitle: undefined,
+  search: false,
+  children: undefined,
+};
+
+TopNav.propTypes = {
+  active: PropTypes.number,
+  bgColor: PropTypes.string,
+  color: PropTypes.string,
+  company: PropTypes.node,
+  serviceTitle: PropTypes.node,
+  search: PropTypes.node,
+  children: PropTypes.node,
+};
+
+export default TopNav;
```

Comment:
this file is huge, could you make an `atoms` folder containing one glamorous component per file - search-wrapper.js, right-hand-side.js etc. ?

Would make the core functionality a lot easier to read.

Reply:
yeah sure